Add `min_new_tokens` argument in generate() implementation

silverriver opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Fixes #20756 #20814 #20614 (cc @gonced8 @kotikkonstantin)

As many said, it is better to add an argument `min_new_tokens` to the `.generate()` method to limit the length of newly generated tokens. The current parameter `min_length` limits the length of `prompt + newly generated tokens`, not the length of `newly generated tokens`.

It seems that all tests are passed

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

- @gante


<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- text models: @ArthurZucker and @younesbelkada
- vision models: @amyeroberts and @NielsRogge
- speech models: @sanchit-gandhi

Library:

- flax: @sanchit-gandhi
- generate: @gante
- pipelines: @Narsil
- tensorflow: @gante and @Rocketknight1
- tokenizers: @ArthurZucker
- trainer: @sgugger

Integrations:

- deepspeed: @stas00
- ray/raytune: @richardliaw, @amogkam

Documentation: @sgugger and @stevhliu

HF projects:

- accelerate: [different repo](https://github.com/huggingface/accelerate)
- datasets: [different repo](https://github.com/huggingface/datasets)
- diffusers: [different repo](https://github.com/huggingface/diffusers)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Maintained examples (not research project or legacy):

- Flax: @sanchit-gandhi
- PyTorch: @sgugger
- TensorFlow: @Rocketknight1

 -->

silverriver committed:
add min_new_tokens

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_20819). All of your documentation changes will be reflected on that endpoint.

AndreasMadsen committed:
Implement Roberta PreLayerNorm (#20305)

* Copy RoBERTa

* formatting

* implement RoBERTa with prelayer normalization

* update test expectations

* add documentation

* add convertion script for DinkyTrain weights

* update checkpoint repo

Unfortunately the original checkpoints assumes a hacked roberta model

* add to RoBERTa-PreLayerNorm docs to toc

* run utils/check_copies.py

* lint files

* remove unused import

* fix check_repo reporting wrongly a test is missing

* fix import error, caused by rebase

* run make fix-copies

* add RobertaPreLayerNormConfig to ROBERTA_EMBEDDING_ADJUSMENT_CONFIGS

* Fix documentation <Facebook> -> Facebook

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

* fixup: Fix documentation <Facebook> -> Facebook

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

* Add missing Flax header

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

* expected_slice -> EXPECTED_SLICE

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

* update copies after rebase

* add missing copied from statements

* make fix-copies

* make prelayernorm explicit in code

* fix checkpoint path for the original implementation

* add flax integration tests

* improve docs

* update utils/documentation_tests.txt

* lint files

* Remove Copyright notice

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* make fix-copies

* Remove EXPECTED_SLICE calculation comments

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

daquexian committed:
lazy import torch._softmax_backward_data for better compatibility (#20796)

lazy import torch._softmax_backward_data

Signed-off-by: daquexian <daquexian566@gmail.com>

Signed-off-by: daquexian <daquexian566@gmail.com>

younesbelkada committed:
[`Vision`] [Refactor] Initialize weights on the correct place (#20803)

* fix nit

- initialization on `_init_weights`
- fix copies

* add copied from

amyeroberts committed:
Vilt - use image_transforms pad (#20780)

Use image_transforms pad

stas00 committed:
[clip] fix error message (#20818)

* [clip] fix error message

* sync

stanleycai95 committed:
Add model resources for ViT (#20723)

* Set up overall resources documentation structure

* Update vit.mdx

* Removing irrelevant sections on text models

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

* Update vit.mdx

Thomas-MMJ committed:
fix typo output not ouput in bitsandbytes trainer test (#20839)

fix typo output not ouput

typo was causing an error on pytest collection

fzyzcjy committed:
Fix tiny typo (#20841)

* Fix typo

* Update README.md

* Update run_mlm_flax_stream.py

* Update README.md

ydshieh committed:
Remove unused `max_position_embeddings ` in config classes (#20836)

Removed unused max_position_embeddings in config classes

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

sanchit-gandhi committed:
[mBART] fix erroneous italics in docstring (#20835)

* [mBART] fix erroneous italics in docstring

* fix-copies

Rocketknight1 committed:
TF AdamWeightDecay fix for 2.11 (#20848)

* Fix incorrect import for the base optimizer for AdamWeightDecay

* Fix incorrect import for the base optimizer for AdamWeightDecay

ydshieh committed:
remove unused `use_cache` in config classes (#20844)

remove unused use_cache in config classes

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

NielsRogge committed:
[SegFormer] Add support for segmentation masks with one label (#20279)

* Add support for binary segmentation

* Fix loss calculation and add test

* Remove space

* use fstring

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>
Co-authored-by: Niels Rogge <nielsrogge@Nielss-MBP.localdomain>

stevhliu committed:
Clarify `use_fast` parameter in docstring (#20840)

* clarify use_fast parameter

* make style

* remove check frameworks, apply review

sanchit-gandhi committed:
[S2T, Whisper] Add copied from statements (#20787)

* [S2T, Whisper] Add copied from statements

* rebase and fix-copies

stevhliu committed:
Embed circle packing chart for model summary (#20791)

* embed circle packing chart

* trim whitespace from bottom

* explain bubble sizes

younesbelkada committed:
Add BLIP (#20716)

* add new model like

* add v1

* v1

* v1

* vision encoder logits match

* v2

* fix

* add docstring

* CI tests pass

* fix tests

* make fixup

* add to `toctree`

* fix processors

* fix processors

* fix doc

* fill title

* add content doc

* remove from tokenization auto

* fix config

* change order

* add `# Copied from`

* few fixes

- add correct license on modeling text
- remove dummy argument

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* replace name

* refactor a bit

* more refactor

* remove unused arg

* make fixup + remove some `# Adapted from ...`

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* more `# Copied from`

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* now `generate` supports no prefix

* remove `FeatureExtractor`

* fix path

* correct dependency

* fix tests

* few fixes

* add integration tests

* add correct conversion script

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* add `blip` to tokenization auto

* fix docstrings

* fix test + add image

* remove processor from uncorrect place

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* clean up a bit

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* clean pixel mask

* clean pixel mask

* fix `F`

* Update src/transformers/models/blip/modeling_blip.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* fix output

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* fix pad token id

* remove `token_type_ids`

* make fixup

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* make fixup

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* add comments

* Update src/transformers/models/blip/modeling_blip.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* remove `token_type_ids`

* make fixup

* better name

* replace with `image_attention_mask`

* refactor

* make fixup

* better docstring

* replace `answer_xx`

* remove ununsed args

* add `labels`

* add `labels`

* fix processing tests

* make fixup

* make fixup

* put correct repo

* remove `pad`

* remove `crop` and `center_crop`

* Update src/transformers/models/blip/image_processing_blip.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* fix

* remove `size_divisor`

* fix weights `init`

* remove unneeded functions

* add suggestions

* minor changes

- change slow test output for PT 1.13
- docstring order

* replace `feature_extractor` by `image_processor`

* fix doctests

* fix weight init order + add fp16 slow test

* add `blip` to doctest

* add correct repo name and fix test

* Update src/transformers/models/blip/processing_blip.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* fix tests

* use `convert_to_rgb` from `image_transforms`

* make fixup

* fix large loading issue

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

NielsRogge committed:
[Swin2SR] Add doc tests (#20829)

* Fix doc tests

* Use Auto API

* Apply suggestion

* Revert "Apply suggestion"

This reverts commit cd9507a86644b4877c3e4a3d6c2d5919d9272dd7.

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>
Co-authored-by: Niels Rogge <nielsrogge@Nielss-MBP.localdomain>

NielsRogge committed:
[Examples] Update big table (#20845)

Update big table

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MBP.localdomain>

ydshieh committed:
Use `config.num_channels` in CLIP-like modeling files (#20857)

Use config.num_channels in CLIP-like modeling files

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

ValeKnappich committed:
fix past_key_values in GPTNeoXForCausalLM.prepare_inputs_for_generation (#20621)

* fix past_key_values in GPTNeoXForCausalLM.prepare_inputs_for_generation

* fix formatting

gante commented:
(Note: this PR depends on the resolution of https://github.com/huggingface/transformers/issues/20814, so I'm waiting for it before I review this one)

idilsulo committed:
Add visual prompt to processor of CLIPSeg model (#20816)

Adds visual_prompt argument to CLIPSegProcessor to enable image-guided segmentation

MKhalusova committed:
Adding `evaluate` to the list of libraries required in generated notebooks (#20850)

Adding `evaluate` to the list of libraries to be installed for every generated notebook in transformers

ydshieh committed:
Fix past CI by skipping `LevitModelTest.test_problem_types` (#20859)

* Fix past CI

* Fix past CI

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

mht-sharma committed:
Fix whisper export (#20800)

* fix_whisper_export

* update input

* update input

ArthurZucker committed:
Fix doctest (#20843)

* fix doc for generation, dinat, nat and prelayernorm

* style

* update

* fix cpies

* use auto config and auto tokenizer

Co-authored-by: sgugger <sylvain.gugger@gmail.com>

* als modify roberta and the depending models

Co-authored-by: sgugger <sylvain.gugger@gmail.com>

ArthurZucker committed:
Add-warning-tokenizer (#20826)

* add fast not use warning

* update

ydshieh committed:
Update `HubertModelIntegrationTest.test_inference_keyword_spotting` (#20863)

fix ci

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

gante committed:
Generate: post-generate config doctest fix (#20804)

* fix doctests

* revert unwanted change

silverriver commented:
> (Note: this PR depends on the resolution of #20814, so I'm waiting for it before I review this one)

Great, let me know if there are anything I can help.

Commit:
change strings to f-strings in image_processing_utils.py (#20865)

change strings to f-strings

younesbelkada committed:
[`FSMT`] Make it compatible with `xxxForConditionalGeneration` models (#20825)

* add `get_encoder` and `get_decoder`

* add additional kwargs support

* fix condition

* add better checks

* better checks

* fix embed positions

* better test to consider padding

* fix debug statement

* Apply suggestions from code review

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

* add arguments on docstring

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

younesbelkada committed:
[`MobileNet-v2`] Fix ONNX typo  (#20860)

* fix typo `onnx`

* fix test

pacman100 committed:
having new model entries in Hindi for Hindi README (#20869)

BakingBrains committed:
Add Onnx Config for PoolFormer (#20868)

poolformer onnx

Co-authored-by: syed <syed.abdul@sandlogic.com>

Narsil committed:
Adding support for `fp16` for asr pipeline. (#20864)

* Supporting `fp16` for asr pipeline

* Adding test.

* Style.

* Oops.

* Flake8 update ?

* Fixing flake8 ?

* Revert "Flake8 update ?"

This reverts commit 0b917fcb520e5f34d1933d9d37d8f32b64553048.

* Style (acctidentally deleted flake8 F401.)

* Move to a bigger test (no small whisper model, and s2t doesn't seem to
accept torch_dtype=fp16).

Also we need to use a GPU to actually compute on fp16.

* Using BatchFeature capability.

bastings committed:
Add script to convert T5X T5 (v1.0 and v1.1) checkpoints to PyTorch (#20801)

* Add script to convert T5X T5 (v1.0 and v1.1) checkpoints to PyTorch

* Remove unnecessary check and update docstring

* Format docstring

* Fix whitespace in docstring

younesbelkada committed:
Add japanese translation of template (#20870)

* add japanese translation of template

* fix japanese translation

- fix special cases
- fix typos
- manually translate special cases

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

ArthurZucker committed:
[RobertaPreLayernom] Fixes the CI daily test (#20886)

get correct checkpoint

makrai committed:
Fixes typo in the help text for --max_length (#20883)

nathanrs committed:
typo fix (#20891)

younesbelkada committed:
[ `T5`] fix fp16 loading issue (#20878)

* fix fp16 loading issue

* add backward compatibility

* better refactor

* better readability

- remove `force_upcast_dtype` as it is used once
- use `inspect`
- add `TODO`

kamalkraj committed:
Update flan-t5 original model link (#20897)

Update flan-t5.mdx

elisim committed:
fix docs typos in "add_new_model" (#20900)

fix Jupyter typos

ydshieh committed:
[Past CI] 🔥 Leave Past CI failures in the past 🔥  (#20861)

* torch.jit._state

* Fix past CI

* Fix for perceiver

* Fix REALM

* Fix for Bloom

* Fix for SwinMode

* Fix for TrajectoryTransformerModel

* Fix for test_wav2vec2_with_lm

* make style

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

akshaya-a committed:
Avoid collisions in writing metrics via 2 APIs - azureml + mlflow (#20837)

* Avoid collisions in writing metrics via 2 APIs - azureml + mlflow

MLflow tracking API is enabled by default in AzureML and HF MLflow integration is more fully featured. I'd remove the AzureML integration but leaving the current behavior for backwards compatibility (though it should really be removed)

* Trigger CI

gante committed:
Generate: correctly detect default max length (#20911)

correctly detect default max length

aphedges committed:
Remove non-breaking spaces (#20929)

* Remove non-breaking space in comment

It was likely added unintionally.

* Remove remaining non-breaking spaces

HarshTrivedi committed:
Load the state dict on CPU to prevent unnecessary GPU memory surge (#20920)

load the state dict on cpu.

bofenghuang committed:
Fix FP16 inference in TextGenerationPipeline (#20913)

* add torch_dtype attribute to Pipeline

* Use torch_dtype to cast input tensor type in AutomaticSpeechRecognitionPipeline

* Fix code quality

* Add TextGenerationPipeline fp16 test

* Fix code quality

* Remove useless require in tests

Co-authored-by: Nicolas Patry <patry.nicolas@protonmail.com>

Co-authored-by: Nicolas Patry <patry.nicolas@protonmail.com>

IvanLauLinTiong committed:
Remove Bert tokenizer dependency from DistillBert (slow/fast) tokenizers (#20933)

mmcdermott committed:
Adds type checking to PreTrainedConfig. (#20926)

bofenghuang committed:
Fix error message in `WhisperFeatureExtractor` (#20936)

* Fix error message

* Fix code quality

samuelzxu committed:
Fixing DistilBert error message (#20945)

Fixing error message

stas00 committed:
[trainer: `distributed_concat`] ensure `all_gather`'s inputs are contiguous (#20951)

[trainer: distributed_concat] ensure all_gather's input are contiguous

bofenghuang committed:
Add generate kwargs to `AutomaticSpeechRecognitionPipeline` (#20952)

* Add generate kwargs to AutomaticSpeechRecognitionPipeline

* Add test for generation kwargs

bofenghuang committed:
Add generate kwargs to `AutomaticSpeechRecognitionPipeline` (#20952)

* Add generate kwargs to AutomaticSpeechRecognitionPipeline

* Add test for generation kwargs

conan1024hao committed:
update pyknp to rhoknp (#20890)

* update pyknp to rhoknp

* fix linter

* fix linter

* fix linter

* fix linter

* fix linter

* support rhoknp==1.1.0, fix testcase

gante committed:
Generate: TF XLA beam sample (#20927)

* beam sample in beam search

* rag now works with the updated beam search

* delete legacy (non-XLA) generation code related to beam sample

IvanLauLinTiong committed:
Fix T5 docstring (#20957)

Fix start_docstring for deparallelize method

gante committed:
Generate: delete unused TF `_reorder_cache` (#20964)

kotikkonstantin committed:
`MinNewTokensLengthLogitsProcessor` for `.generate` method #20814 (#20892)

* feat: add min new length logit processor

* test: add min new length logit processor

* docs: add MinNewTokensLengthLogitsProcessor

* feat: import MinNewTokensLengthLogitsProcessor

* fix: update pytorch dummy objects

* refactor & fix: rename attributes and var and get rid of dynamic attribute

* tests: align test with new interface

* docs: fix typo

* docs: minor clarification

* Empty-Commit

* empty commit

* run automated quality edits

Co-authored-by: Joao Gante <joao@huggingface.co>

alaradirik committed:
Fix post_process_object_detection method descriptions (#20977)

fix post_process_object_detection descriptions

gante commented on src/transformers/generation/utils.py:
```diff
@@ -1234,6 +1234,17 @@ def generate(
                 "(https://huggingface.co/docs/transformers/main/en/main_classes/text_generation)"
             )
 
+        has_default_min_length = kwargs.get("min_length") is None and generation_config.min_length == 0
```
```suggestion
        has_default_min_length = kwargs.get("min_length") is None and generation_config.min_length is not None
```
(some models can have a default value for min_length that is not 0)

gante commented on src/transformers/generation/utils.py:
```diff
@@ -1234,6 +1234,17 @@ def generate(
             )
 
+        has_default_min_length = kwargs.get("min_length") is None and generation_config.min_length == 0
+        if has_default_max_length and generation_config.min_new_tokens is not None:
+            generation_config.min_length = generation_config.min_new_tokens + input_ids_seq_length
```
We don't need to set `min_length` as we have a dedicated logits processor :)

These lines can be removed and, instead, while setting the logits processor list, we can add a `MinNewTokensLengthLogitsProcessor` as defined in #20892 (e.g. after [this line](https://github.com/huggingface/transformers/blob/305f41e4dea99d9fdc212287263b3748aacc63d3/src/transformers/generation/utils.py#L811))

silverriver replied:
Sure, I will make a commit

gante reviewed (commented):
Added some comments to integrate this changes with the changes from #20892 :)

They are small changes, and we should be able to merge afterward!

NielsRogge committed:
Add GIT (GenerativeImage2Text) (#20295)

* First draft

* Make model instantiation work

* Fix copied from statement

* More fixes

* Add correct output head

* Improve configuration

* Add conversion script

* Improve conversion script

* Remove token_type_ids

* Fix conversion of projection layers

* Convert all weights

* Use cats image

* Make logits match

* Generate caption on cats image

* Add GITProcessor

* Update conversion script

* Add support for more checkpoints

* Fix conversion script

* Add initial tests

* Remove cross-attention

* More improvements

* Remove is_decoder

* Improve model tests

* Improve tests

* Improve model outputs

* Fix model outputs equivalence

* Fix more tests

* Remove unused code

* Use generate to generate text, no use of cache for now

* Use generate more appropriately

* Fix config tests

* Fix style

* Add support for use_cache

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

* Fix style

* Fix GIT vision encoder

* Update README

* Fix integration test

* Set bos and eos token ids

* Improve docs

* Improve code

* Add support for provided attention_mask

* Add copied from statement

* Fix gradient checkpointing test

* Set model_input_names

* Investigate model_input_names

* Remove script

* Fix model inputs

* Fix docstring

* Rename GIT to Git

* Support more models

* Add support for textvqa model

* Add video support

* Extend conversion script for video

* Add support for large variant

* Add support for more models

* Fix config archive map

* Update integration test

* Fix README

* Fix CLIP mean and std

* Update processor

* Fix use_cache for video, thanks @gante

* Remove print statements

* Remove assertion

* Add processor tests

* Fix model_input_names

* Use Auto API for processor

* Fix processor tests

* Fix integration test

* Fix pipeline test

* Make tests faster

* Update conversion script

* Update conversion script

* Convert more checkpoints

* Update conversion script

* Fix typo

* Update docstrings

* Improve code snippets

* Fix doc tests

* Add more code examplesé

* Fix doc tests

* Add integration tests

* Fix unused variable

* revert

* Add GIT to Japanese README

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>
Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>
Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

ydshieh committed:
Remove more unused attributes in config classes (#20858)

Remove more unused attributes in config classes

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

sywangyi committed:
[run_clm example] add torch_dtype option for model load. (#20971)

* [run_clm example] add torch_dtype option for model load.
for BLOOM 175B model. peak memory will reduce about 350G for inference. the weight of BLOOM in model hub is bfloat16

Signed-off-by: Wang, Yi A <yi.a.wang@intel.com>

* add other type in option

* fix style

Signed-off-by: Wang, Yi A <yi.a.wang@intel.com>

long8v committed:
Fix valid ratio for Deformable Detr (#20958)

* fix: valid ratio has right value

* chore: remove unnecessary line

Co-authored-by: Jeongyeon Nam <jy.nam@navercorp.com>

samuelpullely committed:
Enable `decoder_attention_mask` in `generate` function (#20726)

* Enable `decoder_attention_mask` in `generate` function

* Make style corrections

* Run `make repo-consistency`

* Add integration test

akrogager committed:
Ignore errors when deleting old checkpoints in trainer (#20984)

ydshieh committed:
Avoid CI runs under users' own CircleCI personal account (#20981)

* Avoid null CI

* Avoid null CI

* rename

* more clear error message

* Update .circleci/config.yml

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* clean up

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

ydshieh committed:
Fix for LXMERT (#20986)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

alaradirik committed:
Improve OWL-ViT postprocessing (#20980)

* add post_process_object_detection method

* style changes

radcheb committed:
Fix race condition on cleaning checkpoints when save_total_limit set to 1 (#20989)

* Update trainer.py

* fix style

Co-authored-by: Radhwane Chebaane <rchebaane.external@epo.org>

tokestermw committed:
Add custom stop token ids for generation (#20727)

* Add StopIdStoppingCriteria

* add a working test for stop id criteria

* add to global scope

* add stop_ids to generate

* add pipeline test

* use tokenizer encode in test

* add test to generation utils

* reformat

* fixup

* make-fix-copies

* rename to stop_token_id

* use stop_tokens instead

* add to text to text generation

* make fixup

* make repo-consistency

* Add support for list of ints for eos_token_id inside generation/utils.py

* Instead of having if elses, cast the eos_token_id into a List[int]

* Add List[int] support for logits_process.py

* add List[int] for beam_search.py

* add List[int] for forced_eos_token_id

* revert stop token id stopping criteria changes

* make fixup

* fix tests

* add eos_token_id to generation/utils.py and added tests test_utils.py

* add eos_token_id type hints and fix for pad tokens

* add comments

* remove some prints and remove forced false test

* fix

* put back test_stop_sequence_stopping_criteria

* remove unused import and make fixup

* add a none check

* update docstring

* add more docstring for list ints

* make fixup

jongjyh committed:
Add AltCLIP (#20446)

* add altclip

* update

* fix wrong title

* fix the copyright in readme

* add altclip model

* add altclip

* fix test_gradient_checkpointing_enable_disable

* code

* add return class

* add projection_state

* "fix pretrained model bug"

* delete print and fix 2 test instances.

* delete token

* rm xlmr

* one model one file.

* empty commit to trigger CI

* Fix modeling_outputs.py

* Fix __init__

* Fix quality

* Fix modeling file docstring

* Fix README.md

* Fix test file

* add vision model

* empty commit to trigger CI

* fix

* fix

* fix

* fix

* fix

* fix

* fix

* fix

* fix

* del token in mdx file

* fix

* fix

* fix

* remove altrob from test list

* add vision test

* fix fx

* fix

* fix

* fix

* trigger CI

* fix copies

* fix tests

* fix style

* fix quality

* update

* recover import

* recover

* add ,

* recover

* fix copies

* trigger CI

* fix

* some of review

* update

* remove import

* last 2

* fix

* fix style

* fix style

* fix bug

* fix uncomment

* fix

* update

* fix

* second review

* empty commit to trigger CI

* empty commit to trigger CI

* fix position

* fix

* empty commit to trigger CI

* empty commit to trigger CI

* third comment

* Update docs/source/en/model_doc/altclip.mdx

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

* Update docs/source/en/model_doc/altclip.mdx

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

* Update src/transformers/__init__.py

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

* Update src/transformers/models/altclip/configuration_altclip.py

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

* Update src/transformers/models/altclip/modeling_altclip.py

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

* Update src/transformers/models/altclip/processing_altclip.py

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

* Update src/transformers/models/altclip/modeling_altclip.py

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

* fix merge

* fix copies

* update

* update

* empty commit to trigger CI

* fix code example

* empty commit to trigger CI

* fix

* empty commit to trigger CI

* empty commit to trigger CI

Co-authored-by: shunxing1234 <xw747777271@gmail.com>
Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>
Co-authored-by: shunxing1234 <33774367+shunxing1234@users.noreply.github.com>
Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

ArthurZucker committed:
update template (#20885)

* update template

* replace redme entries

* make style

MKhalusova committed:
Add: doc page for the object detection task (#20925)

* Added Object Detection task guide (new branch)

* Polished code examples after running make style

* Update docs/source/en/tasks/object_detection.mdx

Rephrasing suggestion from Sayak

Co-authored-by: Sayak Paul <spsayakpaul@gmail.com>

* Update docs/source/en/tasks/object_detection.mdx

A rephrasing suggestion from Sayak

Co-authored-by: Sayak Paul <spsayakpaul@gmail.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sayak Paul <spsayakpaul@gmail.com>

* Update docs/source/en/tasks/object_detection.mdx

typo

Co-authored-by: Sayak Paul <spsayakpaul@gmail.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sayak Paul <spsayakpaul@gmail.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sayak Paul <spsayakpaul@gmail.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sayak Paul <spsayakpaul@gmail.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Applied reviewers suggestions
>
>
Co-authored-by: sayakpaul <spsayakpaul@gmail.com>
Co-authored-by: sgugger <sylvain.gugger@gmail.com>

* polished code examples

* Added a visualization of the inference result. Slightly changed hyperparameters, and updated the results.

* polished code examples

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* Update docs/source/en/tasks/object_detection.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* Applying Steven's review suggestions

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>
Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* minor punctuation fix

Co-authored-by: Sayak Paul <spsayakpaul@gmail.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>
Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>
Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

long8v committed:
auxiliary_loss works for Deformable Detr (#20959)

fix: auxiliary_loss works

Co-authored-by: Jeongyeon Nam <jy.nam@navercorp.com>

amyeroberts committed:
Update image processor parameters if creating with kwargs (#20866)

* Update parameters if creating with kwargs

* Shallow copy to prevent mutating input

* Pass all args in constructor dict - warnings in init

* Fix typo

alaradirik committed:
Fix bug in segmentation postprocessing (#20198)

* Fix post_process_instance_segmentation
* Add test for label fusing

amyeroberts committed:
Don't call deprecated method (#20904)

idilsulo committed:
Fix model hub link (#20998)

milyiyo committed:
Refactor the function get_results (#20999)

gante committed:
Generate: TF uses `GenerationConfig` as the basis for `.generate()` parametrization (#20994)

stevhliu committed:
Update bug report template (#21004)

add maria to bug report

SD-13 committed:
Remove T5 dependency from mT5 model (#20949)

make mt5 independent from t5

sgugger committed:
Fix repo consistency

stevhliu committed:
Update PR template (#21006)

add maria to pr template

sayakpaul committed:
add: task guide on video classification model fine-tuning. (#20827)

* add: task guide on video classification model fine-tuning.

* apply make style from hf-formatting.

* add: toc entry.

* chore: address PR comments.

Co-authored-by Maria Khalusova

* Reflect Maria's contributions.

Co-authored-by: Maria Khalusova <1065417+MKhalusova@users.noreply.github.com>

* chore: minor correction.

* Apply suggestions from code review

Co-authored-by: Nathan Raw <nxr9266@g.rit.edu>

* PyTorch Video -> PyTorchVideo.

* Apply suggestions from code review

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

* change licensing year.

* minor rewording.

* apply make style.

* address Sylvain's comments.

* replace links.

Co-authored-by: Maria Khalusova <1065417+MKhalusova@users.noreply.github.com>
Co-authored-by: Nathan Raw <nxr9266@g.rit.edu>
Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

gante committed:
Generate: Fix CI related to #20727  (#21003)

ydshieh committed:
Fix (DeepSpeed) docker image build issue (#21002)

* Fix docker image build issue

* remove comment

* Add comment

* Update docker/transformers-pytorch-deepspeed-latest-gpu/Dockerfile

Co-authored-by: Stas Bekman <stas00@users.noreply.github.com>

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>
Co-authored-by: Stas Bekman <stas00@users.noreply.github.com>

dependabot[bot] committed:
Bump gitpython from 3.1.18 to 3.1.30 in /examples/research_projects/decision_transformer (#21010)

Bump gitpython in /examples/research_projects/decision_transformer

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.18 to 3.1.30.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.18...3.1.30)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump gitpython from 3.0.2 to 3.1.30 in /examples/research_projects/distillation (#21011)

Bump gitpython in /examples/research_projects/distillation

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.0.2 to 3.1.30.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.0.2...3.1.30)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

stevhliu committed:
Fix callback docstrings (#21005)

* fix callback docstrings

* format as markdown list

* apply feedback

gante committed:
Generate: post-generate config TF doctest fix  (#21018)

gante committed:
Generate: FLAX infers pad token in its absence and has functional example (#21009)

younesbelkada committed:
[`BLIP`] Fix daily CI failing test (#20877)

sgugger committed:
Make sure dynamic objects can be saved and reloaded (#21008)

* Make sure dynamic objects can be saved and reloaded

* Remove processor test

NielsRogge committed:
[CLIPSeg] Fix integration test (#20995)

Fix integration test

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>

gante committed:
Generate: FLAX uses `GenerationConfig` as the basis for `.generate()` parametrization (#21007)

mpierrau committed:
Added mask_time_prob and mask_time_length arguments to wav2vec2 pretraining script (#20985)

Added mask_time_prob and mask_time_length arguments to wav2vec2 pretraining script and readme - new branch

hvaara committed:
[NumPy] Remove references to deprecated NumPy type aliases (#21022)

[NumPy] Remove references to deprecated NumPy type aliases.

This change replaces references to a number of deprecated NumPy type aliases (np.bool, np.int, np.float, np.complex, np.object, np.str) with their recommended replacement (bool, int, float, complex, object, str).

NumPy 1.24 drops the deprecated aliases, so we must remove uses before updating NumPy.

Co-authored-by: Peter Hawkins <phawkins@google.com>

Co-authored-by: Peter Hawkins <phawkins@google.com>

silverriver committed:
Update src/transformers/generation/utils.py

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

Observer46 committed:
Fix arguments passed to predict function in QA Seq2seq training script (#21026)

fix args passed to predict function

david1542 committed:
Support turning off the model uploading in ClearML (#20969)

* Add support for turning off the model uploading in ClearML

* Add documentation for the CLEARML_LOG_MODEL environment variable

* Adjust new doc addition to the new style

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Co-authored-by: Dudu Lasry <dudu.lasry@viz.ai>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

cceyda committed:
fix parameter name in docstring (#21032)

Bearnardd committed:
fix levit timm conversion file (#20938)

* fix levit timm conversion file

* remove set_defaults

silverriver commented:
Sorry that I have accidently rebased my PR to include the change of #20892. It seems that my operation will link this PR to all pervious commits that have been merged to main.

I think it would be more convenient to make a new PR from the main branch. So I am closing this one.

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_20819). All of your documentation changes will be reflected on that endpoint.

silverriver commented:
I have made a new PR #21044
